internal/tool/prepare.go

```diff
@@ -0,0 +1,134 @@
+package tool
+
+import (
+	"bufio"
+	"os"
+	"path"
+	"strings"
+
+	codacy "github.com/codacy/codacy-engine-golang-seed/v6"
+	"github.com/samber/lo"
+)
+
+const sourceConfigFileName = ".semgrep.yaml"
+
+// TODO: should respect cli flag for docs location
+const rulesDefinitionFileName = "/docs/rules.yaml"
+
+func createConfigFile(toolExecution codacy.ToolExecution) (*os.File, error) {
+
+	if toolExecution.Patterns == nil {
+		return createConfigFileFromScratch(toolExecution)
+	}
+
+	if len(*toolExecution.Patterns) == 0 {
+		return nil, nil
+	}
+
+	// if there are patterns, create a configuration file from them
+	return createConfigFileFromPatterns(toolExecution.Patterns)
+}
+
+func createConfigFileFromScratch(toolExecution codacy.ToolExecution) (*os.File, error) {
```

Comment:
I'm a bit confused by this function because the name is not really accurate. I think it would be simpler to have it's logic in the function above (`newConfigurationFile`).